src/burnchains/burnchain.rs

```diff
@@ -1292,6 +1298,15 @@ impl Burnchain {
                     while let Ok(Some(ipc_header)) = downloader_recv.recv() {
                         debug!("Try recv next header");
 
+                        match should_keep_running {
+                            Some(ref should_keep_running)
+                                if !should_keep_running.load(Ordering::SeqCst) =>
+                            {
+                                break
```

Comment:
Should this cause the `sync_with_indexer()` method to return `burnchain_error::CoordinatorClosed`?  Then, upstream callers (like `BitcoinRegtestController`) don't have to worry about polling `should_keep_running`.

Reply:
Addressed with latest